### ISSUE TITLE
Does DOSBox-X support audio input?

### ISSUE THREAD

**Issue by PeterSwinkels** [joncampbell123/dosbox-x#3022]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar question(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Question

How do I use the microphone in DOSBox-X? Windows 3.11 plays audio but doesn't appear to record while runnning in DOSBox-X.

**Comment by rderooy** [comment]
That is correct, there is currently no support for audio input. The question does surface on occasion, but it requires someone to put in the work to implement the functionality. And there are few legacy DOS and Windows applications that really need it.

**Comment by PeterSwinkels** [comment]
All I know is that basic sound recording isn't that complicated: http://www.o-bizz.de/qbtuts/mallard/sbfaq.htm - however I have no idea how hard it would be to implement in DOSBox-X.

**Comment by Torinde** [comment]
See Issue #1824

**Comment by turboboy215** [comment]
Will this ever be done, or never? It's been several years, and I feel like this should be very high priority. Nobody seems to be interested. If needed, a bounty could be put on the issue. Even just capturing internal audio from the host OS would suffice. I would really like to use some old audio recording/editing software. WinUAE has an option for sampler input. This might also be helpful: http://www.o-bizz.de/qbtuts/mallard/sbfaq.htm Here is another program that can be tested: http://eyecandyarchive.com/Color%20Fusion/

**Comment by turboboy215** [comment]
Also, would this help? https://github.com/tonioni/WinUAE/blob/master/sampler.cpp

**Comment by rderooy** [comment]
"very high priority", that would depend on your perspective. Again as mentioned above it depends on developer interest.

**Comment by joncampbell123** [comment]
At the moment the only audio input supported by DOSBox-X are audio generator functions to generate silence, hiss, or a 1KHz test tone for any program that attempts to record from the Sound Blaster emulation. Actual audio input not yet implemented.

**Comment by Torinde** [comment]
@PeterSwinkels, as far as I understand - Jon Campbell and the other contributors did all of what we see here in their free time. 
I'm grateful for that and would advise to refrain from complaining about speed of development. Consider any speed above zero as a bonus.
Also, the project here seems quite accommodating for input by anyone, so if you know how to make that code, I would assume your PR will be accepted swiftly.
Wishes prioritization - I assume for that the comment is useful, e.g. to help gauging relevance of the various requests, getting specific software/use-case examples, etc.

**Comment by joncampbell123** [comment]
> > "very high priority", that would depend on your perspective. Again as mentioned above it depends on developer interest.
> 
> If the developer want to make his programs as useful as possible to others it is important to listen to the wishes of potential users.

I have a job too, and sometimes (such as the last week), I do not have enough time to work on DOSBox-X as much as I'd like.

**Comment by PeterSwinkels** [comment]
Dear mr. Campbell,

My apologies for sounding impatient, demanding and ungrateful. I am well aware of how much time and effort a large project like this and I realize that perhaps I should show a little more appreciation for all the hard work you put into this during your spare time. At no cost no less. DOSBox-X is one of the best and most feature-rich DOSBox-X forks out there and so far I personally really enjoyed using it for playing old games and retro-programming in MS-DOS and Windows 3.x. Thank you for all the hard work you put into DOSBox-X so far.

sincerely,
Peter Swinkels

**Comment by meldro364-glitch** [comment]
Yes, it does now! 🎤

PR #6041 implements real audio input support via Windows WASAPI. Set `recording source=microphone` in the `[sblaster]` section and DOSBox-X will capture from your host microphone.

Tested with Creative Parrot (Direct ADC) and custom DMA recording programs - both work great!
